### ISSUE TITLE
manual `!RangeInclusive::contains` fails for float with trailing dot

### ISSUE THREAD

**Issue by ZuseZ4** [rust-lang/rust-clippy#6583]
I just learned the hard way that floats are normally written as ```x.0```, instead of just ```x.```.
However, clippy should probably be able to handle both ways?

I tried this code:

```rust
// lr is of type f32
if lr < 0. || lr > 1. {
```

I received the following lint:
```rust
warning: manual `!RangeInclusive::contains` implementation
  --> src/rl/algorithms/q_learning.rs:48:12
   |
48 |         if lr < 0. || lr > 1. {
   |            ^^^^^^^^^^^^^^^^^^ help: use: `!(0...=1.).contains(&lr)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains

```
Following the hint lent to a compiler error:
```rust
error: unexpected token: `...`
  --> src/rl/algorithms/q_learning.rs:48:15
   |
48 |         if !(0...=1.).contains(&lr) {
   |               ^^^
   |
help: use `..` for an exclusive range
   |
48 |         if !(0..=1.).contains(&lr) {
   |               ^^
help: or `..=` for an inclusive range
   |
48 |         if !(0..==1.).contains(&lr) {
   |               ^^^
```
Apparently it does not recognize (0...) as (0.)(..), but instead as (0)(...).
Removing one of those three dots leads to an error since he does recognize the second value as a float.
```rust
error[E0308]: mismatched types
  --> src/rl/algorithms/q_learning.rs:48:18
   |
48 |         if !(0..=1.).contains(&lr) {
   |                  ^^ expected integer, found floating-point number

error[E0277]: can't compare `{integer}` with `f32`
  --> src/rl/algorithms/q_learning.rs:48:22
   |
48 |         if !(0..=1.).contains(&lr) {
   |                      ^^^^^^^^ no implementation for `{integer} < f32` and `{integer} > f32`
   |
   = help: the trait `std::cmp::PartialOrd<f32>` is not implemented for `{integer}`

```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (e1884a8 2020-12-29)
- `rustc -Vv`:
  ```
rustc 1.49.0 (e1884a8e3 2020-12-29)
binary: rustc
commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
commit-date: 2020-12-29
host: x86_64-unknown-linux-gnu
release: 1.49.0
  ```

**Comment by giraffate** [comment]
It seems to be fixed at https://github.com/rust-lang/rust-clippy/pull/6320.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2bac8ef58ed354ee86e406a597f56187
